ReSharper.FSharp/src/FSharp.Psi.Features/src/Daemon/Analyzers/LambdaAnalyzer.fs

```diff
@@ -84,8 +86,30 @@ type LambdaAnalyzer() =
             | x -> hasMatches, i = pats.Count, x
 
         compareArgsRec expr 0 null
-
-    let isApplicable (expr: IFSharpExpression) =
+    
+    let isLambdaApplicable (lambda: ILambdaExpr) =
+        let lambda = lambda.IgnoreParentParens()
+        let appTuple = TupleExprNavigator.GetByExpression(lambda)
+
+        let appArg = if isNotNull appTuple then appTuple.IgnoreParentParens() else lambda
+        let app = PrefixAppExprNavigator.GetByArgumentExpression(appArg)
+
+        let isСonvertibleToDelegate =
+            isNotNull app && isNotNull app.Reference &&
+            match app.Reference.GetFSharpSymbol() with
+            | :? FSharpMemberOrFunctionOrValue as m ->
+                m.IsMember &&
+                let lambdaPos = if isNotNull appTuple then appTuple.Expressions.IndexOf(lambda) else 0
+                let args = m.CurriedParameterGroups
+                if args.[0].Count <= lambdaPos then false else
```

Comment:
It doesn't seem to support curried methods:

```fsharp
type T() = 
     static member M () (_: Delegate1) = ()

T.M () (fun x -> x)
```

Reply:
Lambdas in such methods cannot be implicitly converted to delegates.

Reply:
Lambdas in such methods cannot be implicitly converted to a delegate.

Reply:
OK, cool, thanks. It probably worths adding a test case, just in case if something is changed in our tree or in FCS data in future, and we want to make sure we still report the suggestion here.